trade_remedies_public/config/middleware.py

```diff
@@ -92,7 +92,9 @@ def public_request(self, request):
     def __call__(self, request, *args, **kwargs):
         if request.session and request.session.get("token") and request.session.get("user"):
             back_link_url = request.META.get("HTTP_REFERER", reverse("dashboard"))
-            if resolve(request.path_info).url_name in back_link_url:
+            if not resolve(request.path_info).url_name:
```

Comment:
```suggestion
         if resolved_url := resolve(request.path_info).url_name:
             if resolved_url in back_link_url:
                 back_link_url = reverse("dashboard")
```

Doing it with 2 if statements forces django to evaluate `resolve(request.path_info).url_name` twice